### PR TITLE
Implement Weapon types

### PR DESCRIPTION
**Pull request by maikkundev**
# Resolves

This pull request resolves issue #27.

### CODE REVIEW

File: utils/PlayerMoveAttack.png
Comment (actuallylost) [inline utils/PlayerMoveAttack.png:1]: Change the png name so we dont get confused

File: weapons/weapon-type.gd
Comment (actuallylost) [inline weapons/weapon-type.gd:1]: add the `_init` func to the end of the script, follow what entity-type's script does.

File: weapons/weapon.gd [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class_name Weapon
+extends Entity
+
+# The weapon's state
+enum State {
+  IDLE, ATTACK
+}
```
Comment (actuallylost) [inline weapons/weapon.gd:7]:
```suggestion
```
The weapon does not have a state.

File: weapons/weapon.gd [hunk 2]
```diff
@@ -0,0 +1,44 @@
+class_name Weapon
+extends Entity
```
Comment (actuallylost) [inline weapons/weapon.gd:2]:
```suggestion
class_name Weapon
extends WeaponEntity
```
Entities are for living things, WeaponEntity is what you need here.

File: weapons/weapon.gd [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	# update local state
+	state = to as State
+
+	# update weapon sprite
+  # TODO
+	# match state:
+	# 	State.IDLE:
+	# 			sprite.texture = sprite_idle
+	# 	State.ATTACK:
+	# 			sprite.texture = sprite_attack
```
Comment (actuallylost) [inline weapons/weapon.gd:41]:
```suggestion
```
Weapon does not have states.

**Review by actuallylost — CHANGES_REQUESTED** [review]

File: weapons/weapon.gd (hunk 3 above)
Comment (maikkundev) [reply weapons/weapon.gd:41]: wont we have different weapon sprites depending if the player is attacking or not?

Comment (actuallylost) [reply weapons/weapon.gd:41]: It's a player sprite animation, the weapon's asset doesn't change during it

**Review by actuallylost — APPROVED** [review]
Looks good to me, awesome job!

**Pull request merged**